src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      addProperties(state, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      removeProperties(state, action.keysToRemove);
+    } else if (action.type === 'clear') {
+      for (const prop in state) {
+        delete state[prop];
+      }
+    }
+  }
+}
+
```

Comment:
Use switch statement if you have limited amount of conditions. (also from the checklist)